docs/interactions/Slash_Commands.md

```diff
@@ -0,0 +1,841 @@
+
+Using your favorite security library, you **must validate the request each time to you receive an interaction**. If the signature fails validate, respond with a `401` error code. Here's a couple code examples:
+
+```js
+const nacl = require('tweetnacl');
+
+// Your public key can be found on your application in the Developer Portal
+const PUBLIC_KEY = 'APPLICATION_PUBLIC_KEY';
+
+const signature = req.get('X-Signature-Ed25519');
+const timestamp = req.get('X-Signature-Timestamp');
+const body = req.rawBody; // rawBody is expected to be a string, not raw bytes
+
+const isVerified = nacl.sign.detached.verify(
+  Buffer.from(timestamp + body),
+  Buffer.from(signature, 'hex'),
+  Buffer.from(PUBLIC_KEY, 'hex')
+);
+
+if (!isVerified) {
+  return res.status(401).end('invalid request signature');
+}
+ ```
+
+```py
+from nacl.signing import VerifyKey
+from nacl.exceptions import BadSignatureError
+
+# Your public key can be found on your application in the Developer Portal
+PUBLIC_KEY = 'APPLICATION_PUBLIC_KEY'
+
+verify_key = VerifyKey(bytes.fromhex(PUBLIC_KEY))
+
+signature = request.headers["X-Signature-Ed25519"]
+timestamp = request.headers["X-Signature-Timestamp"]
+body = request.data
+
+try:
+    verify_key.verify(f'{timestamp}{body}'.encode(), bytes.fromhex(signature))
+except BadSignatureError:
+    abort(401, 'invalid request signature')
+```
+
+If you are not properly validating this signature header, we will not allow you to save your interactions URL in the Dev Portal. We will also do automated, routine security checks against your endpoint, including purposefully sending you invalid signatures. If you fail the validation, we will remove your interactions URL in the future and alert you via email and System DM.
+
+We highly recommend checking out our [Community Resources](#DOCS_TOPICS_COMMUNITY_RESOURCES/interactions) and the two libraries found there. They not only provide typing for Interactions data models, but also include decorators for API frameworks like Flask and Express to make validation easy.
+
+## Subcommands and Subcommand Groups
+
+> warn
+> Currently, subcommands and subcommand groups all appear at the top level in the command explorer. This may change in the future to include them as nested autocomplete options.
+
+For those developers looking to make more organized and complex groups of commands, look no further than subcommands and groups.
+
+**Subcommands** organize your command by **specifying actions within a command or group**.
+
+**Subcommand Groups** organize your **subcommands** by **grouping commands by similar action or resource within a command**.
+
+These are not enforced rules. You are free to use subcommands and groups however you'd like; it's just how we think about them.
+
+> danger
+> Using subcommands or subcommand groups will make your base command unusable. You can't access the base `/permissions` if you also have `/permissions add | remove` as subcommands or subcommand groups
```

Comment:
Given it is stated that the UI might change so the subcommands are on a single command, will this still hold true.

Reply:
Technically yes but maybe the wording isn't the best. If you have `/permissions add | remove`, then `/permissions` by itself will never be a valid command. I'll think about the wording.